### PR TITLE
Write to notes user association

### PR DESCRIPTION
**Pull request by steventux**
## Context

As part of associating `VendorApiUser`s to `Note`s, and [once we've added the `user_id` and `user_type` columns](https://github.com/DFE-Digital/apply-for-teacher-training/pull/6340) we need to start to populate the `user` association (as well as the `provider_user` association)

## Changes proposed in this pull request

- Adds polymorphic association `user` to `Note`
- Amends `NewNoteForm` to write to the `user` association

## Link to Trello card

https://trello.com/c/7VrvjyP5/4701-api-v11-write-to-noteuser-association-when-creating-notes

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -336,6 +336,8 @@ en:
               too_long: "The note must be %{count} characters or fewer"
             provider_user:
               blank: Missing provider_user
```
Comment (despo) [inline config/locales/en.yml:338]: Can we remove this

File: spec/factories/note.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@
   factory :note do
     application_choice
     provider_user
+    user { create(:provider_user) }
```
Comment (despo) [inline spec/factories/note.rb:5]: can we add some randomness to this and generate notes with both provider_users and vendor_api users.

**Review by despo — APPROVED** [review]
Left some minor feedback but happy to approve!

**Comment by steventux** [comment]
Thanks for review @despo - have amended/rebased as per your comments

**Pull request merged**